fake.go

```diff
@@ -519,6 +521,177 @@ func (fake *Fake) Dump() string {
 	return buf.String()
 }
 
+// extractComment removes comment section from a given string and returns it together with comment string.
+// If a comment wasn't found, an empty comment string will be returned.
+func extractComment(s string) (string, string) {
+	if !strings.Contains(s, "comment") {
+		return s, ""
+	}
+	// find comment section with colon
+	reg := regexp.MustCompile(`comment.*".*" ?;?`)
+	// extract comment section and leave just the string
+	comment := reg.FindStringSubmatch(s)[0]
+	commentReg := regexp.MustCompile(`"[^"]+"`)
+	comment = commentReg.FindStringSubmatch(comment)[0]
+	comment = strings.Trim(comment, "\"")
+	// remove comment section from the initial string
+	res := reg.ReplaceAllString(s, "${1}")
+	return strings.TrimSpace(res), comment
```

Comment:
You should be able to do this with a single regexp... https://github.com/kubernetes-sigs/knftables/commit/fd49603c0a3d6012383657f44cae84781245a1a8 has a function that used to be used when parsing `nft list` output. It was only used for rules, so it assume the comment is the last thing on the line and isn't followed by a semicolon, but you should be able to adjust it